### DISCUSSION TITLE
Getting Access token from token endpoint.

### DISCUSSION THREAD

**Discussion by hrishikesh-pate-vectorsolutions** [zitadel/zitadel#7377] [Q&A] [answered]
Hi all,
I am new to zitadel I have created one instance on zitadel and have created multiple project and in one of the project I have created one Web application and for that particular application Authentication method is "Private Key JWT" I have set one redirect URI, everything regarding the application is configured.
                     I am following the [documentation](https://zitadel.com/docs/apis/openidoauth/endpoints#token_endpoint) provided for getting access token for assertion I am using ["dinochiesa"](https://dinochiesa.github.io/jwt/) as provided in the documentation. For the token endpoint I am making a post request to token endpoint provided in the URLs section in the application.
                     I am passing assertion and grant_type : urn:ietf:params:oauth:grant-type:jwt-bearer  in body and content-type is application/x-www-form-urlencoded it is also specified in the documentation. But the problem is I am getting response as "{
    "error": "invalid_request",
    "error_description": "Errors.User.NotFound"
}".
                     I am using private key from the key which I generate from application providing kid as keyId and Payload is also properly configured. Is there something that I am missing from the documentation? So can you guys help me to get access token for my application.
                     I am able to get access token for service user using private key just FYI if this helps.
                     
Thanks.

**Comment by hifabienne** [comment]
@livio-a can you help here?

**Reply by livio-a** [reply, marked as answer]
hey @hrishikesh-pate-ctr-vectorsolutions 

JWT Profile can be used in two different ways
1. authorization grant (https://zitadel.com/docs/apis/openidoauth/endpoints#jwt-profile-grant)
2. client authentication (for [code exchange grant of web apps](https://zitadel.com/docs/apis/openidoauth/endpoints#authorization-code-grant-code-exchange) and[ token introspection with api apps](https://zitadel.com/docs/apis/openidoauth/endpoints#introspection_endpoint))

From your description, I believe your trying (2) and signing users to your application, which authenticates itself using Private Key JWT, right?
If so, you will need to set `client_assertion_type` and `client_assertion`, resp. change the `grant_type` to `authorization_code`

**Reply by hrishikesh-pate-vectorsolutions** [reply]
Hello @livio-a ,
Apologies for late reply I was able to get  token when I changed grant_type to authorization_code I was able to get the access token and validate it with Introspect endpoint.
Thanks.

**Comment by Tomas809** [comment]
Do I really have to read all this sh*t to find out what a G*d d*ame access token is?
Really? Covid has really upended your game boys.

**Reply by fforootd** [reply]
I am not sure you interpreted the context correctly 😁 This discussion is  not about what the access token are.

We can certainly help you, but I would encourage you to be less offensive in your choice of words 😉 

If you have a specific question please go ahead and open a discussion or join our chat at zitadel.com/chat

**Comment by Tomas809** [comment]
Thank you my frustration got the better of me and for that my appolies
technology is a great tool but if you do not work in that field on a day to
day basis these curveballs only help to add to the anxiety and depression
it creates unfortunatly.

Respectfully,

Thomas 368

**Reply by fforootd** [reply]
No worries this can happen to all of us.

**Comment by Tomas809** [comment]
So....now that I have evoked a respons from the dark matter you will helo
me along in this maze? Lord forbib my neurointerface implode from the
incident.

Thomas368

**Reply by fforootd** [reply]
Lets create a new discussion with you question.

I will be happy to reply there.

**Comment by Tomas809** [comment]
😁- Yes I am trying to figure out my errors when cloning a file. It seems
there are alot
If things I am yet unfamiliar with such as when I first tried cloning a
file I kept getting ask to include " Read me" why? and how exactly do you
include something one would
assume is apart of the
cloning processes
🙄- obviously I'm going to need to read it right? And then 2ndly U was
constantly asked for a token....So I searched hi and hi for a token key of
some sort until I finally figured it out on
my own.....(People there need to stop speaking in Geek and use basic
English as posted in the agreement Rules.) So that being done I deleated my
1st attempt and tried to retrace my steps to clone the files again with the
new information I obtained. The progam said it would notify me by email
when the cloning was completed. Have I done everything correctly.this time
is the question.

Respectfully,

Thoma3s68

**Reply by fforootd** [reply]
Please create a discussion and I will try to help you there.

**Comment by Tomas809** [comment]
So If I am wrong with anything I stated could you please give me the
answers to this photo I
am including to you?

Respectfully,

Thom3a6s8

**Reply by fforootd** [reply]
I do not now what photo you are talking about 😂

Please create a discussion and I will try to help you there.

**Comment by Tomas809** [comment]
My next question is this being processed correctly?

Skip to content

Code
Issues
Preparing your new repository
There is no need to keep this window open, we’ll email you when the import
is done.
Cancel
 Tomas809/D-GitHub-Uber-SD-Filess

 Detecting your project’s version control system…??

Respectfully,

Tho3m6a8s

**Reply by fforootd** [reply]
Please create a discussion and I will try to help you there.

Locking this discussion since it is resolved.
